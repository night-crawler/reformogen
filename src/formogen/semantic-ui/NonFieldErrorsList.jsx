import React from 'react';
import PropTypes from 'prop-types';
import { toPairs } from 'lodash';
import { Grid } from 'semantic-ui-react';

import { ErrorsList } from './ErrorsList';


NonFieldErrorsList.displayName = 'NonFieldErrorsList';
NonFieldErrorsList.propTypes = {
  errors: PropTypes.object,
};
export function NonFieldErrorsList({ errors }) {
  return (
    <Grid className='non-field-errors layout' columns={ 16 }>
      { toPairs(errors).map(([ header, errors ], i) =>
        <Grid.Row key={ i }>
          <div className='sixteen wide column'>
            <ErrorsList header={ header } messages={ errors } />
          </div>
        </Grid.Row>
      ) }
    </Grid>
  );
}
