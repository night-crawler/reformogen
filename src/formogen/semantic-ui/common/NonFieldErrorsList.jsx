import React from 'react';
import PropTypes from 'prop-types';

import _ from 'lodash';

import { Grid } from 'semantic-ui-react';

import ErrorsList from './ErrorsList';


NonFieldErrorsListComponent.displayName = 'NonFieldErrorsList';
NonFieldErrorsListComponent.propTypes = {
    errors: PropTypes.object,
};
function NonFieldErrorsListComponent({ errors }) {
    return (
        <Grid className='non-field-errors layout' columns={ 16 }>
            {
                _(errors).toPairs().value().map(([ header, errors ], i) =>
                    <Grid.Row key={ i }>
                        <div className='sixteen wide column'>
                            <ErrorsList header={ header } messages={ errors } />
                        </div>
                    </Grid.Row>
                )
            }
        </Grid>
    );
}

export default NonFieldErrorsListComponent;
