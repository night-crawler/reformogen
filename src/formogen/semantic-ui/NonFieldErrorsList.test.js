import { NonFieldErrorsList } from './NonFieldErrorsList';


it('<NonFieldErrorsList />', () => {
  const wrapper = shallow(
    <NonFieldErrorsList
      errors={ {
        'header': ['error1', 'error2']
      } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
