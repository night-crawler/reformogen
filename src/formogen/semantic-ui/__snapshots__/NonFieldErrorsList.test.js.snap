// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NonFieldErrorsList /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NonFieldErrorsList
    errors={
      Object {
        "header": Array [
          "error1",
          "error2",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <GridRow>
          <div
            className="sixteen wide column"
          >
            <ErrorsList
              color="red"
              header="header"
              messages={
                Array [
                  "error1",
                  "error2",
                ]
              }
            />
          </div>
        </GridRow>,
      ],
      "className": "non-field-errors layout",
      "columns": 16,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="sixteen wide column"
          >
            <ErrorsList
              color="red"
              header="header"
              messages={
                Array [
                  "error1",
                  "error2",
                ]
              }
            />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ErrorsList
              color="red"
              header="header"
              messages={
                Array [
                  "error1",
                  "error2",
                ]
              }
            />,
            "className": "sixteen wide column",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "red",
              "header": "header",
              "messages": Array [
                "error1",
                "error2",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GridRow>
            <div
              className="sixteen wide column"
            >
              <ErrorsList
                color="red"
                header="header"
                messages={
                  Array [
                    "error1",
                    "error2",
                  ]
                }
              />
            </div>
          </GridRow>,
        ],
        "className": "non-field-errors layout",
        "columns": 16,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="sixteen wide column"
            >
              <ErrorsList
                color="red"
                header="header"
                messages={
                  Array [
                    "error1",
                    "error2",
                  ]
                }
              />
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ErrorsList
                color="red"
                header="header"
                messages={
                  Array [
                    "error1",
                    "error2",
                  ]
                }
              />,
              "className": "sixteen wide column",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "red",
                "header": "header",
                "messages": Array [
                  "error1",
                  "error2",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
